src/mixins/filter.js

```diff
@@ -39,13 +43,12 @@ export default {
                 })
                 .filter(this.rowMatch);
 
-            const exactMatchRows = filteredRows
-                .filter((row) => row._exactMatch);
-
-            if (exactMatchRows.length) {
-                exactMatchRows.forEach((row) => {
-                    this.arrayMove(filteredRows, row._meta.originalIndex, 0);
-                });
+            if (this.exactMatch) {
+                filteredRows
+                    .filter((row) => row._exactMatch)
+                    .forEach((row) => {
+                        this.arrayMove(filteredRows, row._meta.originalIndex, 0);
```

Comment:
By placing every row with an exact match at the top, we change the order of rows. It would be better to retain their relative order.